src/vs/workbench/contrib/terminalContrib/suggest/browser/terminalCompletionService.ts

```diff
@@ -57,31 +63,46 @@ export class TerminalCompletionService extends Disposable implements ITerminalCo
 		});
 	}
 
-	async provideCompletions(promptValue: string, cursorPosition: number, shellType: TerminalShellType): Promise<ISimpleCompletion[] | undefined> {
+	async provideCompletions(promptValue: string, cursorPosition: number, shellType: TerminalShellType, token: CancellationToken, triggeredProviders?: ITerminalCompletionProvider[]): Promise<ISimpleCompletion[] | undefined> {
```

Comment:
Passing in triggeredProviders here is a little confusing. Could we not allow the terminal completion service complete control over what providers are used?

Reply:
I find it confusing to move this code because we only want to request completions when trigger chars are enabled and `!sent` 🤔 

https://github.com/microsoft/vscode/blob/1944b106c3d0d87da8acebe887f7c56b4c267eab/src/vs/workbench/contrib/terminalContrib/suggest/browser/terminalSuggestAddon.ts#L267-L282